compiler/modules/types/list.bal

```diff
@@ -270,15 +270,17 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, Conju
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        foreach int i in 0 ..< len {
-            SemType d = diff(fixedArrayGet(members, i), listMemberAt(nt.members, nt.rest, i));
-            if !isEmpty(cx, d) {
-                FixedLengthArray s = fixedArrayShallowCopy(members);
-                fixedArraySet(s, i, d);
-                if listInhabited(cx, s, rest, neg.next) {
-                    return true;
+        if len != 0 {
+            foreach int i in 0 ..< int:max(members.initial.length(), nt.members.initial.length()) {
```

Comment:
We can remove the `if` and bring that condition into the upper limit of the int-range, I'm not sure which way to go.